sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -233,6 +236,9 @@ private[sql] class SQLConf extends Serializable {
 
   private[spark] def dataFrameSelfJoinAutoResolveAmbiguity: Boolean =
     getConf(DATAFRAME_SELF_JOIN_AUTO_RESOLVE_AMBIGUITY, "true").toBoolean
+
+  private[spark] def dataFrameRetainGroupColumns: Boolean =
+    getConf(DATAFRAME_RETAIN_GROUP_COLUMNS, "true").toBoolean
```

Comment:
Increasingly wondering if dataframe flags should be scoped (eager analysis affects `sql(...)` too and not just dataframe DSL functions).,